syntax = "proto3";
package woops.rpc;

service Comm {
    rpc CheckAlive(CheckAliveRequest) returns (CheckAliveResponse) {}
    rpc Finish(FinishRequest) returns (FinishResponse) {}
    rpc BarrierNotify(BarrierNotifyRequest) returns (BarrierNotifyResponse) {}
    rpc SyncPlacement(SyncPlacementRequest) returns (SyncPlacementResponse) {}
    rpc SyncStorage(SyncStorageRequest) returns (SyncStorageResponse) {}
    rpc Update(stream UpdateRequest) returns (stream UpdateResponse) {}
    rpc Pull(stream PullRequest) returns (stream PullResponse) {}
    rpc Push(stream PushRequest) returns (stream PushResponse) {}
}


message CheckAliveRequest {}

message CheckAliveResponse {
    bool status = 1; 
}

message FinishRequest {}

message FinishResponse {}

message SyncPlacementRequest {}

message SyncPlacementResponse {
    bytes data = 1;
}

message BarrierNotifyRequest {}

message BarrierNotifyResponse {}

message SyncStorageRequest {
    int32 tableid = 1;
    bytes parameter = 2;
}

message SyncStorageResponse {
}

message UpdateRequest {
    int32 tableid = 1;
    int32 iteration = 2;
    bytes data = 3;
}

message UpdateResponse {
}

message PullRequest {
    int32 tableid = 1;
    int32 iteration = 2;
}

message PullResponse {}

message PushRequest {
    int32 tableid = 1;
    int32 iteration = 2;
    bytes data = 3;
}

message PushResponse {}

